CRM/Utils/System/Joomla.php

```diff
@@ -479,6 +487,13 @@ public function setUserSession($data) {
   public function setMessage($message) {
   }
 
+  public function factoryClassName(): string {
+    if (version_compare(JVERSION, '5.0.0', 'ge')) {
```

Comment:
Think this should be if greater than or equal to `4.0.0`? I believe the `J` aliased classes are deprecated / not officially supported in Joomla 4.x?